### DISCUSSION TITLE
How to access data of custom components with .on('event', callback_function) event listeners

### DISCUSSION THREAD

**Discussion by thaije** [zauberzeug/nicegui#1022] [Q&A] [answered]
### Question

In the very helpful example of a custom counter component [see here](https://github.com/zauberzeug/nicegui/blob/main/examples/custom_vue_component/main.py), the current value of the component is updated on click and can be accessed in Python, and shows on click the current counter value in a notification. 

However, when I try to access this value with a click listener that is added via a separate `counter.on("click", cb)` function, I can't seem to get this to work. What would be the way to do this? 
For example, changing the `main.py` of the counter example to: 
``` python
#!/usr/bin/env python3
from counter import Counter

from nicegui import ui

ui.markdown('''
#### Try the new click counter!

Click to increment its value.
''')
with ui.card():
    counter = Counter('Clicks', on_change=lambda msg: ui.notify(f'The value changed to {msg["args"]}.'))
    
# prints "Click! {'isTrusted': True, '_vts': 1686642331948}", and not "Click! >value of counter<"
def cb(msg):
    print("Click!", msg["args"])

# the new click listener 
counter.on("click", cb)

ui.button('Reset', on_click=counter.reset).props('small outline')

ui.run(port=1234)
```

The output in the `cb` on click function returns: `Click! {'isTrusted': True, '_vts': 1686642331948}`, while I would expect the counter value. 
Does anyone know how I can access this? Any help is appreciated!

**Comment by falkoschindler** [comment, marked as answer]
Hi @thaije!
The JavaScript "click" event might contain event arguments like clientX, clientY [and so on](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event), but it has nothing to do with our custom "change" event emitted in counter.js.

To get specific arguments of some custom event, you can use the `args` parameter like this:
```py
counter.on("click", cb, args=['clientX', 'clientY'])
```

But to get the `value` of the counter, you need to use the "change" event (the value is sent by default):
```py
counter.on("change", cb)
```

By the way: We're planning to improve the event registration with `.on()` in version 1.3, since it isn't as nice as it could be (#583, #672).

**Reply by thaije** [reply]
Thanks for the quick reply and the solution! I can confirm that it works :smile:
